modules/core/src/ocl.cpp

```diff
@@ -7050,12 +7056,12 @@ const char* convertTypeStr(int sdepth, int ddepth, int cn, char* buf)
         (ddepth == CV_16S && sdepth <= CV_8S) ||
         (ddepth == CV_16U && sdepth == CV_8U))
     {
-        snprintf(buf, buf_max, "convert_%s", typestr);
+        snprintf(buf, buf_size, "convert_%s", typestr);
     }
     else if( sdepth >= CV_32F )
-        snprintf(buf, buf_max, "convert_%s%s_rte", typestr, (ddepth < CV_32S ? "_sat" : ""));
+        snprintf(buf, buf_size, "convert_%s%s_rte", typestr, (ddepth < CV_32S ? "_sat" : ""));
     else
-        snprintf(buf, buf_max, "convert_%s_sat", typestr);
+        snprintf(buf, buf_size, "convert_%s_sat", typestr);
```

Comment:
It would be nice to check `snprintf` return values.
We don't really want to go forward with truncated messages ("fail fast" is preferable).

Reply:
And do what if truncation occurs? Return NULL? None of the callers of `convertTypeStr` appear ready to get NULL.

Anyway, truncation is much better than a buffer overrun, which is what would happen before this PR.

Reply:
> Return NULL

No, my proposal to use `CV_Assert()` or `CV_Check*()` macros (they work in both debug/release modes). These calls raise C++ exception on a problem.

---

> Anyway, truncation is much better than a buffer overrun, which is what would happen before this PR.

Agreed. Thank you for the fix!